### PR TITLE
Build NuGetProvider from PackageManagement NuGet package

### PR DESCRIPTION
**Pull request by brywang-msft**
Build change is so that NuGetProvider doesn't take a dependency on PackageManagement repo during its own build.

### CODE REVIEW

**Comment by brywang-msft** [comment]
https://ci.appveyor.com/project/jianyunt/nugetprovider/build/1.0.98

File: vs-csproj/NugetLightProvider.csproj [hunk 1]
```diff
@@ -63,47 +63,104 @@
...
+	<Compile Include="..\Utility\PackageBaseConverter.cs" />
+	<Compile Include="..\Utility\PackageDependencyConverter.cs" />
+	<Compile Include="..\Utility\PackageFilterUtility.cs" />
+    <Compile Include="..\Utility\PackageUtility.cs" />
+    <Compile Include="..\Utility\ProgressTracker.cs" />
+	<Compile Include="..\Utility\PublicObjectView.cs" />
+    <Compile Include="..\Utility\TaskGroup.cs" />
+    
+    
+    <!--<Compile Include="..\..\oneget\src\Microsoft.PackageManagement\Providers\Inbox\Common\Utility\PathUtility.cs" />
```
Comment (jianyunt) [inline vs-csproj/NugetLightProvider.csproj]: Should we just remove them instead of commenting them out?

File: pack.ps1 [hunk 2]
```diff
@@ -0,0 +1,146 @@
+<#
+Create a NuGet package from an output location
```
Comment (jianyunt) [inline pack.ps1:2]: does the pack.ps1 get called in post build?

File: pack.ps1 [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+        }
+
+        if ((Test-Path -Path $tempDir) -and ($tries -gt 0)) {
+            Start-Sleep -Milliseconds (100 * $tries)
+            $tries = $tries - 1
+        }
+    }
+
+    if (Test-Path -path $tempDir) {
+        Write-Error "Failed to remove temp directory: $tempDir"
```
Comment (jianyunt) [inline pack.ps1]: Whether we can delete tempdir or not won't impact our packing, right? If so I recommend write-warning.

File: releasenotes.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
+# 3.0.0.2
+## What's new?
+- Packaged as a NuGet package on https://powershell.myget.org/gallery/powershell-core
+
```
Comment (jianyunt) [inline releasenotes.md]: The NuGet packages for both OneGet nugetprovider and PackageManagement are stored in https://powershell.myget.org/gallery/powershell-core  feed?

File: nuget.config [hunk 5]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+  <packageSources>
+    <add key="PowerShell-Module-Feed" value="https://powershell.myget.org/F/powershellmodule/api/v2" />
```
Comment (jianyunt) [inline nuget.config:4]: Here we are using https://powershell.myget.org/F/powershellmodule/api/v2 and I saw somewhere we are using https://powershell.myget.org/gallery/powershell-core ?

File: appveyor.yml [hunk 6]
```diff
@@ -11,12 +11,18 @@ init:
...
+  - ps: pushd c:\projects\NuGetTest\NuGetProvider
+  - ps: c:\projects\NuGetTest\NuGetProvider\Generate-Resources.ps1
+  - ps: $null = Invoke-WebRequest https://dist.nuget.org/win-x86-commandline/latest/nuget.exe -OutFile c:\projects\NuGetTest\NuGetProvider\nuget.exe
+  - ps: pushd c:\projects\NuGetTest\NuGetProvider\vs-csproj
+  - ps: c:\projects\NuGetTest\NuGetProvider\nuget.exe restore -PackagesDirectory c:\projects\NuGetTest\NuGetProvider\vs-csproj\packages
+  - ps: popd
+  - ps: popd
 build_script:
   - msbuild "c:\projects\NuGetTest\NuGetProvider\vs-csproj\NugetLightProvider.csproj" /verbosity:normal
 after_build:
```
Comment (jianyunt) [inline appveyor.yml:28]: Should we update our smoke test cases to make sure we are using the latest nuget provider?

File: pack.ps1 (hunk 2 above)
Comment (brywang-msft) [reply pack.ps1:2]: NuGet packing should be done as part of the release process.

File: vs-csproj/NugetLightProvider.csproj (hunk 1 above)
Comment (brywang-msft) [reply vs-csproj/NugetLightProvider.csproj]: b461de0

File: pack.ps1 (hunk 3 above)
Comment (brywang-msft) [reply pack.ps1]: 2592ee5

File: releasenotes.md (hunk 4 above)
Comment (brywang-msft) [reply releasenotes.md]: c44c680

File: nuget.config (hunk 5 above)
Comment (brywang-msft) [reply nuget.config:4]: c44c680

File: appveyor.yml (hunk 6 above)
Comment (brywang-msft) [reply appveyor.yml:28]: Not sure what you mean. Can you elaborate / rephrase please?

Comment (jianyunt) [reply appveyor.yml:28]: There are some tests under nugetprovider.

**Review by jianyunt — APPROVED** [review]

**Pull request merged**